packages/cli/src/deploy/createPrepareDeploy.ts

```diff
@@ -1,24 +1,25 @@
-import { DeterministicContract, Library, LibraryPlaceholder, salt } from "./common";
+import { DeterministicContract, LibraryPlaceholder, salt } from "./common";
 import { spliceHex } from "@latticexyz/common";
 import { Hex, getCreate2Address, Address } from "viem";
+import { LibraryMap, getLibraryKey } from "./getLibraryMap";
 
 export function createPrepareDeploy(
   bytecodeWithPlaceholders: Hex,
   placeholders: readonly LibraryPlaceholder[],
 ): DeterministicContract["prepareDeploy"] {
-  return function prepareDeploy(deployer: Address, libraries: readonly Library[]) {
+  return function prepareDeploy(deployer: Address, libraryMap: LibraryMap) {
     let bytecode = bytecodeWithPlaceholders;
     for (const placeholder of placeholders) {
-      const library = libraries.find((lib) => lib.path === placeholder.path && lib.name === placeholder.name);
+      const libraryKey = getLibraryKey(placeholder);
+      const library = libraryMap[libraryKey];
```

Comment:
Previously we were doing a linear search to find the matching library every time is function was called, now we're just looking up the matching library in constant time. Doesn't make a big difference with small numbers of libraries but compounds with many libraries.